libraries/ts-command-line/src/CommandLineParameterProvider.ts

```diff
@@ -143,21 +187,23 @@ abstract class CommandLineParameterProvider {
     return key;
   }
 
-  private _createParameter(definition: IBaseCommandLineDefinition,
-                           argparseOptions?: argparse.ArgumentOptions,
-                           key?: string,
-                           /* tslint:disable-next-line:no-any */
-                           converter?: IConverterFunction<any>): CommandLineParameter<any> {
+  private _createParameter<TValue>(
+    definition: IBaseCommandLineDefinition<TValue>,
+    argparseOptions?: argparse.ArgumentOptions,
+    key?: string,
```

Comment:
>key?: string, [](start = 4, length = 13)

This parameter is not needed. It is part of definition.

Reply:
It's only part of `IKeyedCommandLineDefinition`, not of `IBaseCommandLineDefinition`.

---
In reply to: [148402706](https://github.com/Microsoft/web-build-tools/pull/423#discussion_r148402706) [](ancestors = 148402706)